_posts/2015-11-11-full-stack-es6-with-webpack-and-babel.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+title: "On the way to full stack ECMAScript 6"
+date: "2015-11-16 12:00:00"
+icon: code
+categories: tech-stories
+authors: ["Paolo"]
+---
+
+**JavaScript is moving fast these days.** With all the languages out there compiling to it, JS itself is making a big leap forwards with the standardisation of its next version called _ECMAScript 6_ (or _ES6_, or _ECMAScript 2015_, or _ES2015_. [Naming is probably the hardest programming related task](https://twitter.com/a_ramella/status/656522461480099840), and if you think that _ECMAScript sounds like a skin disease_, you're [in good company with the inventor of JavaScript himself](https://mail.mozilla.org/pipermail/es-discuss/2006-October/000133.html)).
+
+In this post, I want to share things we've learned during our ongoing adoption of the new language features.
+
+## JS @epages
+My team and I are working on an upcoming part of the ePages software that provides an easy to use drag-and-drop editor for the layout and appearance of an ePages shop, and serves the actual shop web site by displaying all product and editorial data in that layout.
+
+Under the hood this means aggregating and manipulating JSON data from various REST resources, and providing a live preview of the result in the layout edit mode. Both are tasks JavaScript is really good at, so my team does **JS all the way** from the (Mongo) database through the (Node) server down to the browser.
+
+## JavaScript modules revisited
+One of our pains as JavaScript developers has always been **the lack of a module system**. Node.js introduced the _CommonJS_ approach consisting of `module.exports` and `require('module')`, which works well on the server but not on the client. That's because in the browser, we have network latency, and thus loading a module is an asynchronous operation. When we started our project, _RequireJS_ solved this for us as elegant as it can get without introducing a compile step, using its _AMD (asynchronous module definition)_ syntax consisting of `define(['module'], function (module) {...});`.
```

Comment:
using its _AMD (asynchronous module definition)_ syntax, consisting of `define(['module'], function (module) {...});`.